test/fixtures/s3-fixture/src/main/java/fixture/s3/S3HttpHandler.java

```diff
@@ -201,10 +201,24 @@ public void handle(final HttpExchange exchange) throws IOException {
                 exchange.sendResponseHeaders((upload == null ? RestStatus.NOT_FOUND : RestStatus.NO_CONTENT).getStatus(), -1);
 
             } else if (request.isPutObjectRequest()) {
-                final Tuple<String, BytesReference> blob = parseRequestBody(exchange);
-                blobs.put(request.path(), blob.v2());
-                exchange.getResponseHeaders().add("ETag", blob.v1());
-                exchange.sendResponseHeaders(RestStatus.OK.getStatus(), -1);
+                // a copy request is a put request with a copy source header
```

Comment:
Do we not have to do anything special here for the multipart copy implementation?

Reply:
hmm yes, I must be missing test coverage here.

Reply:
I would expect `org.elasticsearch.repositories.blobstore.testkit.analyze.S3RepositoryAnalysisRestIT` to cover this, but not every time

Reply:
I added a test override to drop the multipart copy threshold down to 1MB (symmetric with the multipart upload override). On my laptop I found that was still too large to hit it reliably so I tested it locally at 10K, at which point testWriteMaybeCopyRead would fail about 10% of the time with the original code here. It passes with https://github.com/elastic/elasticsearch/pull/125737/commits/cb2e324951095cfee60ddd8863d3d325cf35b799 at the 10k threshold (which is technically illegal - S3 documents a minimum of 5MB - but I think for testing it's fine).